### PR TITLE
Code documentation

### PR DESCRIPTION
**Pull request by Namiii**
**TASK**=> T17086

**SUMMARY**
In this PR we have added documentation to most of the classes in the SDK.

### CODE REVIEW

File: sdk/src/main/java/co/omise/android/CardNumber.java [hunk 1]
```diff
@@ -3,7 +3,7 @@
 import co.omise.android.models.CardBrand;
 
 /**
- * CardNumber provides helper methods for working with Personal Account Numbers.
+ * CardNumber provides helper methods for working with personal account numbers.
```
Comment (danfowler) [inline sdk/src/main/java/co/omise/android/CardNumber.java]:
```suggestion
 * CardNumber provides helper methods for working with primary account numbers (credit and debit card numbers).
```

File: sdk/src/main/java/co/omise/android/api/EndPoint.kt [hunk 2]
```diff
@@ -4,15 +4,14 @@ import okhttp3.CertificatePinner
 import okhttp3.HttpUrl
 import java.util.*
 
-
 /**
- * Endpoints encapsulates information about a particular Omise API endpoint.
- * Currently there exists 2 endpoints, the API and the VAULT.
- *
+ * Endpoint is a base class that can be used to encapsulates information
```
Comment (danfowler) [inline sdk/src/main/java/co/omise/android/api/EndPoint.kt]:
```suggestion
 * Endpoint is a base class that can be used to encapsulate information
```

File: sdk/src/main/java/co/omise/android/api/EndPoint.kt [hunk 3]
```diff
@@ -61,6 +59,9 @@ abstract class Endpoint {
     }
 
     companion object {
+        /**
+         * Class containing all the information "VAULT" endpoint.
```
Comment (danfowler) [inline sdk/src/main/java/co/omise/android/api/EndPoint.kt]: information (for the?) "VAULT" endpoint?

File: sdk/src/main/java/co/omise/android/api/EndPoint.kt [hunk 4]
```diff
@@ -72,6 +73,9 @@ abstract class Endpoint {
             }
         }
 
+        /**
+         * Class containing all the information "API" endpoint.
```
Comment (danfowler) [inline sdk/src/main/java/co/omise/android/api/EndPoint.kt]: information (for the?) "API" endpoint?

File: sdk/src/main/java/co/omise/android/api/Request.kt [hunk 5]
```diff
@@ -5,14 +5,19 @@ import okhttp3.HttpUrl
 import okhttp3.RequestBody
 
 /**
- * Request class acts as a holder class that encapsulate the information regarding which of the Omise APIs
- * the user wants to access and pass any additional data needed for that api request
+ * Request class acts as a holder class that encapsulate the information regarding which
```
Comment (danfowler) [inline sdk/src/main/java/co/omise/android/api/Request.kt]:
```suggestion
 * Request class acts as a holder class that encapsulates the information regarding which
```

File: sdk/src/main/java/co/omise/android/api/Request.kt [hunk 6]
```diff
@@ -5,14 +5,19 @@ import okhttp3.HttpUrl
...
+ * of the Omise APIs the user wants to access and pass any additional data needed for that api request
```
Comment (danfowler) [inline sdk/src/main/java/co/omise/android/api/Request.kt]:
```suggestion
 * of the Omise APIs the user wants to access and pass any additional data needed for that API request
```

File: sdk/src/main/java/co/omise/android/models/CardParam.kt [hunk 7]
```diff
@@ -2,6 +2,12 @@ package co.omise.android.models
 
 import com.fasterxml.jackson.annotation.JsonProperty
 
+/**
+ * CardParam holds all the information about a Card that can be supplied to create
```
Comment (danfowler) [inline sdk/src/main/java/co/omise/android/models/CardParam.kt]: This model doesn't seem to support new billing address fields: https://www.omise.co/tokens-api#create

File: sdk/src/main/java/co/omise/android/models/Serializer.kt [hunk 8]
```diff
@@ -18,7 +18,10 @@ import java.io.IOException
 import java.io.InputStream
 import java.io.OutputStream
 
-
+/**
+ * Serializer is the custom serializer used in the SDK to deserialize API
```
Comment (danfowler) [inline sdk/src/main/java/co/omise/android/models/Serializer.kt]:
```suggestion
 * Serializer is the custom serializer used in the SDK to serialize and deserialize API
```

File: sdk/src/main/java/co/omise/android/AuthorizingPaymentURLVerifier.kt [hunk 9]
```diff
@@ -3,7 +3,10 @@ package co.omise.android
 import android.content.Intent
 import android.net.Uri
 
-
+/**
+ * AuthorizingPaymentURLVerifier is a utility class that verifies the correctness and integrity
+ * of a authorizing url that is returned as part of creating a Charge through Omise API.
```
Comment (danfowler) [inline sdk/src/main/java/co/omise/android/AuthorizingPaymentURLVerifier.kt]:
```suggestion
 * of an authorizing URL that is returned as part of creating a Charge through the Omise API.
```

File: sdk/src/main/java/co/omise/android/AuthorizingPaymentURLVerifier.kt [hunk 10]
```diff
@@ -42,6 +45,13 @@ class AuthorizingPaymentURLVerifier {
         expectedReturnURLPatterns = returnURLStringPatterns.map { Uri.parse(it) }
     }
 
+    /**
+     * verifyURL checks a supplied Uri to see if it matches the expected scheme, host and
```
Comment (danfowler) [inline sdk/src/main/java/co/omise/android/AuthorizingPaymentURLVerifier.kt]:
```suggestion
     * verifyURL checks a supplied URL to see if it matches the expected scheme, host and
```

File: sdk/src/main/java/co/omise/android/AuthorizingPaymentURLVerifier.kt [hunk 11]
```diff
@@ -42,6 +45,13 @@ class AuthorizingPaymentURLVerifier {
...
+     * URL path.
```
Comment (danfowler) [inline sdk/src/main/java/co/omise/android/AuthorizingPaymentURLVerifier.kt]:
```suggestion
     * path.
```

File: sdk/src/main/java/co/omise/android/AuthorizingPaymentURLVerifier.kt [hunk 12]
```diff
@@ -42,6 +45,13 @@ class AuthorizingPaymentURLVerifier {
...
+     *
+     * @param uri Uri that needs to be verified.
```
Comment (danfowler) [inline sdk/src/main/java/co/omise/android/AuthorizingPaymentURLVerifier.kt]:
```suggestion
     * @param uri URL that needs to be verified.
```

File: sdk/src/main/java/co/omise/android/AuthorizingPaymentURLVerifier.kt [hunk 13]
```diff
@@ -54,6 +64,13 @@ class AuthorizingPaymentURLVerifier {
         return false
     }
 
+    /**
+     * verifyExternalURL checks if a supplied Uri, from an external source, matches
```
Comment (danfowler) [inline sdk/src/main/java/co/omise/android/AuthorizingPaymentURLVerifier.kt]:
```suggestion
     * verifyExternalURL checks if a URL supplied from an external source matches
```

File: sdk/src/main/java/co/omise/android/AuthorizingPaymentURLVerifier.kt [hunk 14]
```diff
@@ -54,6 +64,13 @@ class AuthorizingPaymentURLVerifier {
...
+     * the expected scheme, host and URL path.
```
Comment (danfowler) [inline sdk/src/main/java/co/omise/android/AuthorizingPaymentURLVerifier.kt]:
```suggestion
     * the expected scheme, host and path.
```

File: sdk/src/main/java/co/omise/android/AuthorizingPaymentURLVerifier.kt [hunk 15]
```diff
@@ -54,6 +64,13 @@ class AuthorizingPaymentURLVerifier {
...
+     *
+     * @param uri Uri that needs to be verified.
```
Comment (danfowler) [inline sdk/src/main/java/co/omise/android/AuthorizingPaymentURLVerifier.kt]:
```suggestion
     * @param uri URL that needs to be verified.
```

File: sdk/src/main/java/co/omise/android/api/Configurer.kt [hunk 16]
```diff
@@ -7,27 +7,27 @@ import okhttp3.Response
 
 
 /**
- * Configurer handles HTTP requests configuration. You can use the [.configure] method
- * to setup your own [okhttp3.OkHttpClient] and avoid using the [Client] directly, for example.
+ * Configurer is used in the [Client] class to handle HTTP request configuration.
+ * The [Configurer.configure] function can also be used to directly configure a
```
Comment (danfowler) [inline sdk/src/main/java/co/omise/android/api/Configurer.kt]:
```suggestion
 * The [Configurer.configure] function can also be used to directly configure
```

File: sdk/src/main/java/co/omise/android/api/EndPoint.kt [hunk 17]
```diff
@@ -4,15 +4,14 @@ import okhttp3.CertificatePinner
...
 import java.util.*
 
-
 /**
- * Endpoints encapsulates information about a particular Omise API endpoint.
- * Currently there exists 2 endpoints, the API and the VAULT.
- *
+ * Endpoint is a base class that can be used to encapsulates information
+ * about a particular Omise API endpoint. Currently there are two endpoint
+ * classes, "API" and "VAULT".
```
Comment (danfowler) [inline sdk/src/main/java/co/omise/android/api/EndPoint.kt]:
```suggestion
 * classes: "API" and "VAULT".
```

File: sdk/src/main/java/co/omise/android/api/RequestBuilder.kt [hunk 18]
```diff
@@ -31,33 +31,36 @@ abstract class RequestBuilder<T : Model> {
...
+     * @return HTTP method as a string.
      */
     open fun method(): String {
         return GET
     }
 
     /**
-     * Abstract method that needs to be implement by all children of this class to provide API Path
+     * Abstract method that needs to be implemented by all children of this class
+     * to provide API Path.
```
Comment (danfowler) [inline sdk/src/main/java/co/omise/android/api/RequestBuilder.kt]:
```suggestion
     * to provide API path.
```

File: sdk/src/main/java/co/omise/android/api/RequestBuilder.kt [hunk 19]
```diff
@@ -31,33 +31,36 @@ abstract class RequestBuilder<T : Model> {
...
      *
-     * @return the url path as [HttpUrl]
+     * @return the URL path as [HttpUrl].
      */
     protected abstract fun path(): HttpUrl
 
     /**
-     * Additional parameters for the request, which is null by default for requests that do not accept params (eg: GET)
+     * Additional parameters for the request, which is null by default for requests
+     * that do not accept params (eg: GET).
```
Comment (danfowler) [inline sdk/src/main/java/co/omise/android/api/RequestBuilder.kt]:
```suggestion
     * that do not accept params (e.g. GET).
```

File: sdk/src/main/java/co/omise/android/models/CardBrand.kt [hunk 20]
```diff
@@ -6,6 +6,16 @@ import kotlinx.android.parcel.IgnoredOnParcel
 import kotlinx.android.parcel.Parcelize
 import java.util.regex.Pattern
 
+
+/**
+ * CardBrand holds information about all card brands recognized in the SDK.
+ *
+ * @param name Card brand name.
+ * @param patternStr Pattern that can be recognized the brand's card numbers.
```
Comment (danfowler) [inline sdk/src/main/java/co/omise/android/models/CardBrand.kt]:
```suggestion
 * @param patternStr Pattern that can be used to recognize the brand's card numbers.
```

File: sdk/src/main/java/co/omise/android/models/Model.kt [hunk 21]
```diff
@@ -6,6 +6,10 @@ import com.fasterxml.jackson.annotation.JsonTypeInfo
 import com.fasterxml.jackson.databind.annotation.JsonTypeIdResolver
 import org.joda.time.DateTime
 
+/**
+ * Model is a base class that all model classes are extended from and contains fields
```
Comment (danfowler) [inline sdk/src/main/java/co/omise/android/models/Model.kt]:
```suggestion
 * Model is a base class from which all model classes are extended and contains fields
```

File: sdk/src/main/java/co/omise/android/models/PaymentMethod.kt [hunk 22]
```diff
@@ -4,6 +4,12 @@ import com.fasterxml.jackson.annotation.JsonProperty
 import kotlinx.android.parcel.Parcelize
 import org.joda.time.DateTime
 
+/**
+ * PaymentMethod contains all the information regarding a payment method available to the
+ * user. PaymentMethod list can be found in the [Capability] object.
+ *
+ * @see <a href="https://www.omise.co/capability-api">Capabilities API</a>
```
Comment (danfowler) [inline sdk/src/main/java/co/omise/android/models/PaymentMethod.kt]:
```suggestion
 * @see [Capabilities API](https://www.omise.co/capability-api)
```

Comment (danfowler) [inline sdk/src/main/java/co/omise/android/models/PaymentMethod.kt]: This was a Markdown link for the Tokens documentation above

File: sdk/src/main/java/co/omise/android/ui/OmiseListFragment.kt [hunk 23]
```diff
@@ -17,7 +17,9 @@ import androidx.recyclerview.widget.RecyclerView.NO_POSITION
 import co.omise.android.R
 import kotlinx.android.synthetic.main.fragment_list.recycler_view
 
-
+/**
+ * OmiseListFragment is the base class for all list based UI classes.
```
Comment (danfowler) [inline sdk/src/main/java/co/omise/android/ui/OmiseListFragment.kt]:
```suggestion
 * OmiseListFragment is the base class for all list-based UI classes.
```

File: sdk/src/main/java/co/omise/android/AuthorizingPaymentURLVerifier.kt (hunk 9 above)
[the remark this answers is not in the document]
Comment (Namiii) [reply sdk/src/main/java/co/omise/android/AuthorizingPaymentURLVerifier.kt]: I thought about that as well, it's good to hear it from you too. Will make the change 👌

File: sdk/src/main/java/co/omise/android/api/EndPoint.kt (hunk 3 above)
Comment (Namiii) [reply sdk/src/main/java/co/omise/android/api/EndPoint.kt]: Correct 👍

File: sdk/src/main/java/co/omise/android/models/PaymentMethod.kt (hunk 22 above)
Comment (Namiii) [reply sdk/src/main/java/co/omise/android/models/PaymentMethod.kt]: 👌

File: sdk/src/main/java/co/omise/android/models/CardParam.kt (hunk 7 above)
Comment (Namiii) [reply sdk/src/main/java/co/omise/android/models/CardParam.kt]: Great point, will need to be added either in this version or another in the near future 👌

File: sdk/src/main/java/co/omise/android/AuthorizingPaymentURLVerifier.kt (hunk 9 above)
[the remark this answers is not in the document]
Comment (Namiii) [reply sdk/src/main/java/co/omise/android/AuthorizingPaymentURLVerifier.kt]: Seems like you have already covered this in your suggestions, so no need for any further changes 🙏

**Pull request merged**